airflow/utils/task_group.py

```diff
@@ -645,6 +659,20 @@ def get_current_task_group(cls, dag: DAG | None) -> TaskGroup | None:
 
         return cls._context_managed_task_group
 
+    @classmethod
+    def add_task(cls, task: DependencyMixin) -> None:
+        """Add the task to the current TaskGroup if not already there."""
+        from airflow.utils.edgemodifier import EdgeModifier
+
+        if isinstance(task, TaskGroup) or isinstance(task, EdgeModifier):
+            return
+        task_ = cast(DAGNode, task)
```

Comment:
I don’t think these `isinstance` checks (here and in the outer `set_taskgroup_ctx_dependencies`) is exhausive enough. Instead of trying to do all the checks, it may be better to rethink how this operation would make sense to all `DependencyMixin` decendants and use polymorphism to handle this instead.

Reply:
I have tried to work out this, but it seems it won't be possible for this use case. For example, `TaskGroup` is a subclass of `DependencyMixin` but here I don't want to add a task group into `TaskGroup.add`. This means I need to filter it out. TaskGroup.add can accept a TaskGroup since a TaskGroup is a subclass of DAGNode but for this use case, I don't want a TaskGroup to be added into it. The `isinstance` on EdgeModifier seems inevitable too since I don't also want an EdgeModifier in TaskGroup.add. 
I'm open to ideas on how to do it.

Reply:
Maybe we can add a method on `DependencyMixin` called something like `add_to_task_group(self, tg)` (name suggestions welcomed). A `TaskGroup` or `EdgeModifier` can implement this as a single `pass` (i.e. do nothing at all) and other subclasses can implement it as appropriate. Alternatively `get_nodes_to_add_to_task_group()` where `TaskGroup` returns None, operator returns itself, and XComArg returns `self.operator` etc.

Reply:
Still on this

Reply:
Done. WDYT?